PWGHF/vertexingHF/AliAnalysisTaskSEXic0Semileptonic.cxx

```diff
@@ -501,10 +538,50 @@ void AliAnalysisTaskSEXic0Semileptonic::UserExec(Option_t*)
 	if (inputHandler->IsEventSelected() & AliVEvent::kHighMultV0)  fHistos->FillTH1("hEventNumbers", "HMV0", 1);
 	if (inputHandler->IsEventSelected() & AliVEvent::kHighMultSPD) fHistos->FillTH1("hEventNumbers", "HMSPD", 1);
 
+    //Judge if this event is INEL > 0, for data - kimc
+	//2nd updated at June 22, position moved to check INEL>0 effect unbiased
+    fIsINELLgtZERO = false;
+    if (IsMC==false && AliPPVsMultUtils::IsINELgtZERO(fEvt)==true) fIsINELLgtZERO = true;
+
+	//INEL>0 effect check histograms
+	if (IsMC == false)
+	{
+        for (int a=0; a<2; a++)
+        {
+            const char* fHistName = Form("hINELLgt0_%i", a);
+            if (a==1 && fIsINELLgtZERO==false) continue;
+
+            if (inputHandler->IsEventSelected() & AliVEvent::kINT7)
+            {
+                fHistos->FillTH1(fHistName, 0.);
+                if (fCentrality >=  0.0 && fCentrality <= 100.0) fHistos->FillTH1(fHistName, 1.);
+                if (fCentrality >=  0.1 && fCentrality <=  30.0) fHistos->FillTH1(fHistName, 2.);
+                if (fCentrality >= 30.0 && fCentrality <= 100.0) fHistos->FillTH1(fHistName, 3.);
+            }
+            if (inputHandler->IsEventSelected() & AliVEvent::kHighMultV0)
+            {
+                fHistos->FillTH1(fHistName, 4.);
+                if (fCentrality >= 0.0 && fCentrality <= 0.1) fHistos->FillTH1(fHistName, 5.);
+            }
+        }//a, trig and trig+INEL>0
+	}//data only
+
 	//*****************************************************
 
-	//Reject pile-up events
-	if (!IsMC && fEvt->IsPileupFromSPD(3.,0.8,3.,2.,5.)) return;
+	//Reject pile-up events: updated at Sep.17, 2021. Referred below link
+	//https://twiki.cern.ch/twiki/bin/viewauth/ALICE/AliDPGtoolsPileup
+	Int_t nContributor = -999;
+	if (IsPA) nContributor = 5; //For pA, but there's a NOTE at above link - PLEASE CHECK
+	else
+	{
+		if      ( fNSPDTracklets < 20 ) nContributor = 3; //Conventional, valid for low multiplicity pp
+		else if ( fNSPDTracklets < 50 ) nContributor = 4; //20 ~ 50
+		else nContributor = 5; //>= 50, valid for high multiplicity pp and p-Pb
+	}
+	if (nContributor == -999) { cout <<"ERROR in pileup cut!\n"; assert(false); }
+	if (!IsMC && fEvt->IsPileupFromSPD(nContributor, 0.8, 3., 2., 5.)) return;
```

Comment:
We typically use the multi-vertexer algorithm for pileup rejection in pp (see https://github.com/alisw/AliPhysics/blob/b2904d50626090b88c2c5b445a7a0d758e7eb7ac/OADB/AliAnalysisUtils.h#L38), is there a reason for using the SPD based algorithm instead for this analysis?

Reply:
Hi @fgrosa, We used the SPD based rejection as this link (https://twiki.cern.ch/twiki/bin/viewauth/ALICE/AliDPGtoolsPileup). I missed the MV based rejection at that time. 
As you mentioned, I will replace the pileup rejection using MV based, and will check it how much difference will be shown.

Reply:
Thanks a lot!